src/main/java/algorithms/curated170/medium/CampusBikes2.java

```diff
@@ -0,0 +1,89 @@
+package algorithms.curated170.medium;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class CampusBikes2 {
+
+      private int dist(int wPos, int bPos) {
+            return Math.abs(workers[wPos][0] - bikes[bPos][0]) + Math.abs(workers[wPos][1] - bikes[bPos][1]);
+      }
+
+      private int targetBitmask;
+      int[][] workers, bikes;
+      int[][] dp;
+
+      public int assignBikes(int[][] workers, int[][] bikes) {
+            dp = new int[workers.length][1<<bikes.length];
+            
+            this.workers = workers;
+            this.bikes = bikes;
+            return DFS(0, (1<<bikes.length) - 1);
+      }
+
+      private int DFS(int worker, int mask)
+      {
+            if(worker == workers.length)
+            {
+                  return 0;
+            }
+            if(dp[worker][mask] != 0)
+            {
+                  return dp[worker][mask];
+            }
+            int ans = Integer.MAX_VALUE;
+            for(int j = 0; j<bikes.length; j++)
+            {
+                  if(( mask & (1<<j) )!= 0)
+                  {
+                        ans = Math.min(ans, dist(worker, j) + DFS(worker+1, mask^(1<<j)));
+                  }
+            }
+            return dp[worker][mask] = ans;
+      }
+      public static void main(String[] args) {
+
+            var solution = new CampusBikes2();
+            int[][] workers = leetCodeIntegerGridConverter("[[0,0],[1,1],[2,0],[3,4]]");
+            int[][] bikes = leetCodeIntegerGridConverter("[[1,0],[2,2],[2,3],[4,0]]");
+
+            int sol = solution.assignBikes(workers, bikes);
+            System.out.println(sol);
+      }
+
+      public static int[][] leetCodeIntegerGridConverter(String arr) {
```

Comment:
You can create a package named "utils" or "util" and move this under a custom class.
You can name the class something like LeetCodeUtil as we can maintain and enhance it in the future.